lib/benchee/configuration.ex

```diff
@@ -333,6 +331,28 @@ defmodule Benchee.Configuration do
     DeepMerge.deep_merge(%{formatter_options: formatter_options}, config)
   end
 
+  # backwards compatible formatter definition without leaving the burden on every formatter
+  @module_to_formatter_options_key %{
+    Benchee.Formatters.Console => :console,
+    Benchee.Formatters.CSV => :csv,
+    Benchee.Formatters.JSON => :json,
+    Benchee.Formatters.HTML => :html
+  }
+  defp formatter_options_to_tuples(config) do
+    formatters =
+      Enum.map(config.formatters, fn formatter ->
+        compatible_formatter_options_key = @module_to_formatter_options_key[formatter]
+
+        if compatible_formatter_options_key do
+          {formatter, config.formatter_options[compatible_formatter_options_key]}
+        else
+          formatter
+        end
```

Comment:
I would write this as:
```elixir
Enum.map(config.formatters, fn
  Benchee.Formatters.Console -> {formatter, config.formatter_options[:console]}
  Benchee.Formatters.CSV -> {formatter, config.formatter_options[:csv]}
  Benchee.Formatters.JSON -> {formatter, config.formatter_options[:json]}
  Benchee.Formatters.HTML -> {formatter, config.formatter_options[:html]}
  formatter -> formatter
end)
```

You could also get a bit fancy with elixir and do something like:

```elixir
update_in(config, [Access.key(:formatters), Access.all()], fn 
```

Reply:
Hm I sort of liking to keep the relationship as data and have the code be more general. However, this is highly temporary code where I hope I can delete it soooooonnnn. Then that version makes a lot more sense as it's way more condensed.

The `update_in` seems like... woooahhhh. I need to read up on `Access` module it seems.

Thanks a ton Michał! :pray: